### PR TITLE
Add a "admin user generate-access-token" subcommand

### LINKED ISSUE #17721: Add a way to generate access token from the CLI for automation purpose
### Feature Description

It would be useful to me to be able to get a access token for a user in a way that would be easier to use by a external program (eg, automation with ansible or something else).

The current options to get a token are:
- use the API (so I need a access token to generate a access token)
- use the web interface (not very automation friendly)
- generate a new user using gitea admin CLI, and parse the message, which is a bit fragile and only work with a new user.

So I would propose to add a subcommand to the admin user command:
```
$ gitea admin user generate-access-token --help
NAME:
   Gitea admin user generate-access-token - Generate a access token for a specific user

USAGE:
   Gitea admin user generate-access-token [command options] [arguments...]

OPTIONS:
   --username value, -u value     Username
   --token-name value, -t value   Token name (default: "gitea-admin")
   --raw                          Display only the token value
   --custom-path value, -C value  Custom path file path (default: "/tmp/custom")
   --config value, -c value       Custom configuration file path (default: "/tmp/custom/conf/app.ini")
   --version, -v                  print the version
   --work-path value, -w value    Set the gitea working path (default: "/tmp")
   

DEFAULT CONFIGURATION:
     CustomPath:  /tmp/custom 
     CustomConf:  /etc/gitea/app.ini
     AppPath:     /tmp/gitea
     AppWorkPath: /tmp
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by mscherer**
Fixes #17721

### CODE REVIEW

File: cmd/admin.go [hunk 1]
```diff
@@ -538,6 +560,46 @@ func runDeleteUser(c *cli.Context) error {
...
+	}
+
+	ctx, cancel := installSignals()
+	defer cancel()
+
+	if err := initDB(ctx); err != nil {
+		return err
+	}
+
+	if err := storage.Init(); err != nil {
```
Comment (wxiaoguang) [inline cmd/admin.go]: I think we only need db access, so do we really need to init storage?

File: cmd/admin.go [hunk 2]
```diff
@@ -538,6 +560,46 @@ func runDeleteUser(c *cli.Context) error {
...
+	}
+
+	if err := models.NewAccessToken(t); err != nil {
+		return err
+	}
+
+	if c.Bool("raw") {
+		fmt.Printf("%s\n", t.Token)
+	} else {
+		fmt.Printf("Access token was successfully created... %s\n", t.Token)
```
Comment (wxiaoguang) [inline cmd/admin.go]:
```suggestion
		fmt.Printf("Access token was successfully created: %s\n", t.Token)
```

File: cmd/admin.go [hunk 3]
```diff
@@ -538,6 +560,46 @@ func runDeleteUser(c *cli.Context) error {
...
+	}
+
+	return nil
+
```
Comment (wxiaoguang) [inline cmd/admin.go]:
```suggestion
```

File: cmd/admin.go (hunk 1 above)
Comment (mscherer) [reply cmd/admin.go]: Let me remove and try it

Comment (mscherer) [reply cmd/admin.go]: Ok, it wasn't needed, I changed the PR , thanks for noticing it.

**Review by wxiaoguang — APPROVED** [review]

**Comment by zeripath** [comment]
OK, so one last thing before we put this in:

* Are we sure that it's OK for the administrator to create tokens for users?

I guess if the admin has access the command line and database they can do what they like so ...

Also it appears that if the admin logs in with BasicAuth and sets Sudo they can create tokens for any user from the API.

**Review by singuliere — APPROVED** [review]

**Comment by singuliere** [comment]
@mscherer could you please answer @zeripath question?

**Comment by mscherer** [comment]
@singuliere I think one of us misunderstood the question. I do not think that's being asked to me. I proposed the feature and exposed my rationale on the linked bug, so yes I am ok.

The way I understand it, it is asked to others devs (I mean, the "we" part kinda indicate to be aimed at the committers group to me). 

So maybe it should be rephrased as "can you (singuliere) answer to @zeripath question ?" :p  

I assume that you still have the same need as I have wrt ansible automation for a gitea installation, and I guess you are still using what was [discussed before](https://forum.forgefriends.org/t/unattended-gitea-installation-from-the-cli/195). I added that feature because i want to automate woodpecker deployment (which is I think also on the table for you), and that would requires a token to use the API. Hence this PR.

**Comment by zeripath** [comment]
I think the question was: Are we sure that we want to do this?

I guess as the admin user can already do this through Sudo it's not a new thing so we can merge.

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**